server/plugin/api.go

```diff
@@ -265,6 +265,12 @@ func (p *Plugin) handleCreateSubscription(w http.ResponseWriter, r *http.Request
 		return
 	}
 
+	if channelAccessErr := p.CheckIfUserCanAccessChannel(body.ChannelID, mattermostUserID); channelAccessErr != nil {
+		p.API.LogError(constants.ErrorCreateSubscription, "Error", channelAccessErr.Error())
+		p.handleError(w, r, &serializers.Error{Code: http.StatusForbidden, Message: "You are not allowed to create subscription for the provided channel ID"})
```

Comment:
i think we should use the status code present in the error instead of using forbidden status code

Reply:
status code here from the MM is `404` which is correct from MM's perspective but from this plugin API perspective `404` would be misleading as while creating a subscription nothing is missing but the user has provided a resource to which he doesn't has access

Reply:
are we sure the API used in the function always returns a 404 code? what if there is a case of some internal server error